### PR TITLE
feat: Created Vehicle Interior Doctype

### PR DESCRIPTION
**Pull request by MAunZaidi**
closes: https://github.com/ParaLogicTech/hyundai_south_print/issues/16

### CODE REVIEW

File: erpnext/vehicles/doctype/vehicle/vehicle.json [hunk 1]
```diff
@@ -750,11 +751,20 @@
    "label": "Reserved By Sales Person",
    "no_copy": 1,
    "options": "Sales Person"
+  },
+  {
+   "allow_in_quick_entry": 1,
+   "fieldname": "interior",
+   "fieldtype": "Link",
+   "in_preview": 1,
```
Comment (SaiFi0102) [inline erpnext/vehicles/doctype/vehicle/vehicle.json:759]: Do not add in preview

File: erpnext/stock/doctype/item/item.json [hunk 2]
```diff
@@ -57,6 +57,7 @@
   "quotation_validity_days",
   "column_break_35",
   "vehicle_colors",
+  "vehicle_interior",
```
Comment (SaiFi0102) [inline erpnext/stock/doctype/item/item.json:60]: fieldname should be plural for table fields

**Review by SaiFi0102 — COMMENTED** [review]
Missing feature:
- Link query for filtering allowed interior colors
- Checkbox in Item to make interior color mandatory in VBO

File: erpnext/stock/doctype/item/item.json (hunk 2 above)
Comment (MAunZaidi) [reply erpnext/stock/doctype/item/item.json:60]: Should I change it everywhere?

Comment (SaiFi0102) [reply erpnext/stock/doctype/item/item.json:60]: Do not change, I will change it myself

**Comment by SaiFi0102** [comment]
Review changes and improvements: https://github.com/ParaLogicTech/erpnext/commit/7240f9f1a1c3af41da3f2196333f81b4f84214d0

**Pull request merged**